### PR TITLE
Fix and test for issue #4117

### PR DESCRIPTION
**Pull request by rwgk**
## Description
See #3861 and #4117.

## Suggested changelog entry:

```rst
TODO
```

### CODE REVIEW

**Comment by rwgk** [comment]
@laramiel thanks for suggesting the test!

(I wasn't getting very lucky looking under https://github.com/wassimj/Topologic)

**Comment by rwgk** [comment]
@henryiii there is a test here.

**Comment by rwgk** [comment]
@Lalaland @laramiel It's interesting that the CI succeeded for #4273, it simply replaces `intrinsic_t<T>` with `T`.

@Lalaland did you come to the idea that `intrinsic_t` is unnecessary by inspecting the calling code?

I almost think it will be best to drop this PR and reopen and merge #4273 (the only hiccup was the unused ctypes import). Although it might be useful to add in the Python part of the test from here, which has a Python override returning `self` (@laramiel's idea). I believe it is redundant for the purpose of exercising the exact core fix, but it is a different situation that we evidently don't cover otherwise, and is very little extra code.

@henryiii for awareness.

BTW: Amazing coincidence that 3 people worked on fixes in 2 PRs at the same time.

**Comment by Lalaland** [comment]
> @Lalaland <https://github.com/Lalaland> did you come to the idea that
intrinsic_t is unnecessary by inspecting the calling code?

I switched it to T just to revert to the old behavior in 2.9

Your version with remove_cv is more correct (and an improvement over 2.9,
as opposed to just fixing the regression).

I can quickly reopen #4273 with the needed changes (remove_cv and returning
self).

**Comment by Lalaland** [comment]
Don't actually have permission to reopen PRs, so created #4275

File: tests/test_virtual_functions.cpp [hunk 1]
```diff
@@ -230,6 +230,35 @@ inline int test_override_cache(std::shared_ptr<test_override_cache_helper> const
 // are rather long).
 void initialize_inherited_virtuals(py::module_ &m);
 
+namespace issue_4117 { // Broken by PR #3861
+
+class Animal {
+public:
+    virtual ~Animal() {}
```
Comment (Skylion007) [inline tests/test_virtual_functions.cpp:237]:
```suggestion
    virtual ~Animal() = default;
```

**Comment by rwgk** [comment]
Closing, superseded by #4275

**Comment by rwgk** [comment]
FWIW / future reference: I rebased and squashed my branch after #4275 was merged, JIC we want to resurrect the test later. I also verified that the test succeeds as expected.

**Pull request closed without merging**